recipe/build-mpi.sh

```diff
@@ -92,14 +101,18 @@ if [[ "$target_platform" == linux-* ]]; then
 fi
 
 ./configure --prefix=$PREFIX \
+            --enable-fast=O3 \
+            --enable-romio \
             --disable-dependency-tracking \
             --enable-cxx \
             --enable-fortran \
             --enable-f08 \
             --with-wrapper-dl-type=none \
             --disable-opencl \
-            --with-device=ch4 \
-            || cat config.log
+            --with-hwloc=$PREFIX \
+             --enable-nemesis-shm-collectives \
+            $build_with_ucx \
+            || (cat config.log; exit 1)
```

Comment:
nit: I prefer adding `set -ex` at the top of the script and keep the install command here simple. We used to have issues with CI when the output log was too long. This would double the log size.

Reply:
This is here because the output of configure is often insufficient to diagnose problems, producing output like "configure failed, FEATURE unavailable. See config.log for why...". Note that this `cat` only happens when configure fails. This has no effect on the output of successful builds.